src/tools/rest-backend/model_user.hpp

```diff
@@ -0,0 +1,298 @@
+#ifndef ELEKTRA_REST_MODEL_USER_HEADER_GUARD
+#define ELEKTRA_REST_MODEL_USER_HEADER_GUARD
+
+#include <map>
+#include <string>
+#include <utility>
+
+#include <boost/algorithm/string/compare.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+
+#include "config.hpp"
+#include "exceptions.hpp"
+#include "kdb_includes.hpp"
+
+namespace kdbrest
+{
+
+namespace model
+{
+
+typedef bool (*comp_func) ();
+
+class User : public kdb::Key
+{
+
+public:
+	/**
+                 * Constructs an User object based on a kdb::Key object.
+                 * It is an alternative for an upwards cast.
+                 * @param key A kdb::Key object
+                 */
+	inline User (kdb::Key & k) : kdb::Key (k)
```

Comment:
Interesting to know, wasn't aware of that! Is it ok expressing it explicitely or should I remove it?